include/odp/api/spec/ipsec.h

```diff
@@ -772,6 +773,71 @@ typedef struct odp_ipsec_sa_param_t {
 
 } odp_ipsec_sa_param_t;
 
+/**
+ * IPSEC SA information
+ */
+typedef struct odp_ipsec_sa_info_t {
+	/** Copy of IPSEC Security Association (SA) parameters */
+	odp_ipsec_sa_param_t param;
+
+	/** IPSEC SA direction dependent parameters */
+	union {
+		/** Inbound specific parameters */
+		struct {
+			/** Additional SA lookup parameters. */
+			struct {
+				/** IP destination address (NETWORK ENDIAN) to
+				 *  be matched in addition to SPI value. */
+				uint8_t dst_addr[ODP_IPV6_ADDR_SIZE];
+			} lookup_param;
+
+			/** Antireplay window size
+			 *
+			 * Antireplay window size configured for the SA.
+			 * This value can be different from what application
+			 * had requested.
+			 */
+			uint32_t antireplay_ws;
+
+			/** Antireplay window top
+			 *
+			 * Sequence number representing a recent top of the
+			 * antireplay window.
+			 */
+			uint64_t antireplay_window_top;
+		} inbound;
+
+		/** Outbound specific parameters */
+		struct {
+			/** Sequence number
+			 *
+			 * Sequence number used for a recently processed packet
+			 * by the IPSEC SA. Need not be the latest.
+			 */
+			uint64_t seq_num;
```

Comment:
One more comment on the API: This is not well defined when no packets have yet been processed. Should this be zero then? Or maybe this field could be the sequence number that will be used for the next successfully transformed packet?

Reply:
Yes, agreed that this case should be defined. When no packets have been processed, it should be zero. Shall I update the documentation as follows?:

```
 * Sequence number used for a recently processed packet(need not be the latest)
 * by the IPSEC SA. When no packets have been processed, this value will be zero.
 */
```

Reply:
Space missing here too. Maybe similar rewording here too:

Sequence number used for a recently processed packet. There may be a delay before the SA state is reflected in the value. When no packets have been processed, the value will be zero.

Reply:
Updated the API description as suggested.